internal/docker/docker_test.go

```diff
@@ -251,7 +253,85 @@ func TestDockerRun(t *testing.T) {
 					RepoDir:   "absolute/path/to/repo",
 					ApiRoot:   testAPIRoot,
 				}
-				return d.Configure(ctx, configureRequest)
+				jsonData, _ := json.MarshalIndent(&config.LibraryState{}, "", "  ")
+				jsonFilePath := filepath.Join(configureRequest.RepoDir, config.LibrarianDir, config.ConfigureResponse)
+				os.WriteFile(jsonFilePath, jsonData, 0644)
+				_, err := d.Configure(ctx, configureRequest)
+				defer os.RemoveAll(configureRequest.RepoDir)
```

Comment:
This should go higher up, not sure exactly where, but definitely not just before you return an error.
(If you want to do this here, just do it, don't defer it.)